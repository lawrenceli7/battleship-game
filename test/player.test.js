import { expect, test } from "@jest/globals";
import Gameboard from "../src/js/gameboard";
import Player from "../src/js/player";

test("is either of type human or computer", () => {
  expect(() => {
    new Player("human");
  }).not.toThrow();
  expect(() => {
    new Player("computer");
  }).not.toThrow();
  expect(() => {
    new Player("bird");
  }).toThrow();
  expect(() => {
    new Player("barbell");
  }).toThrow();
});

test("contains its own gameboard", () => {
  const human = new Player("human");
  expect(human.gameboard).toBeInstanceOf(Gameboard);

  const computer = new Player("computer");
  expect(computer.gameboard).toBeInstanceOf(Gameboard);

  expect(computer.gameboard).not.toBe(human.gameboard);

  expect(human).toMatchSnapshot();
  expect(computer).toMatchSnapshot();
});
