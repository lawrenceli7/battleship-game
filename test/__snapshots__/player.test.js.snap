// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contains its own gameboard 1`] = `
Player {
  "gameboard": Gameboard {
    "allShipsSunk": false,
    "board": [
      [
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
      ],
      [
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
      ],
      [
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
      ],
      [
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
      ],
      [
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
      ],
      [
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
      ],
      [
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
      ],
      [
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
      ],
      [
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
      ],
      [
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
      ],
    ],
    "hits": Set {},
    "length": 10,
    "misses": Set {},
    "ships": [],
  },
  "type": "human",
}
`;

exports[`contains its own gameboard 2`] = `
Player {
  "gameboard": Gameboard {
    "allShipsSunk": false,
    "board": [
      [
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
      ],
      [
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
      ],
      [
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
      ],
      [
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
      ],
      [
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
      ],
      [
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
      ],
      [
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
      ],
      [
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
      ],
      [
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
      ],
      [
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
      ],
    ],
    "hits": Set {},
    "length": 10,
    "misses": Set {},
    "ships": [],
  },
  "type": "computer",
}
`;
